### PR TITLE
Add fields retry

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Allow changing all the fields from FuzzedExponentialIntervals in the retry decorator

R: @chamikaramj PTAL

### CODE REVIEW

File: sdks/python/apache_beam/utils/retry.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def no_retries(fun):
 def with_exponential_backoff(
     num_retries=16, initial_delay_secs=5.0, logger=logging.warning,
     retry_filter=retry_on_server_errors_filter,
-    clock=Clock(), fuzz=True):
+    clock=Clock(), fuzz=True, factor=2, max_delay_secs=60 * 60 * 4):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/utils/retry.py]: Please explain these in Args.

Comment (sb2nov) [inline sdks/python/apache_beam/utils/retry.py]: Done

**Comment by sb2nov** [comment]
R: @chamikaramj PTAL

**Comment by chamikaramj** [comment]
LGTM

**Comment by sb2nov** [comment]
@davorbonaci Please merge

**Comment by davorbonaci** [comment]
Done.

**Comment by sb2nov** [comment]
Thanks

**Pull request closed without merging**